Sync boshrelease with latest cg-uaa merge

rogeruiz opened the pull request:
Tasks:
- [x] Update submodules
- [x] Upgrade versions in `config/blob.yml`
- [x] Update blobstore across environments
- [x] Clean up `config/blobs.yml` ( :thinking: )
- [x] Various `packages/` updates
- [x] Various `jobs/` updates

cc: @linuxbozo, @cnelson

rogeruiz committed:
Deleting duplicates

rogeruiz committed:
Update blobs

rogeruiz committed:
Update packages versions

rogeruiz committed:
Bring in cg-uaa upstream merge changes

rogeruiz committed:
Add health_check changes

rogeruiz committed:
Add the log4j.properties changes

rogeruiz committed:
Add the post-start changes

rogeruiz committed:
Add the uaa_ctl changes

rogeruiz commented:
The `Add * changes` commits leveraged this script to perform a `diff` between release job files.

```
#!/bin/bash

set -eu

uaa_theirs=~/Developer/cloud.gov/uaa-release
uaa_ours=~/Developer/cloud.gov/cg-cf-release

# shellcheck disable=SC2045
for file in $(ls $uaa_ours/jobs/uaa/templates)
do
  file_name=$(echo "$file" | cut -d '/' -f 2)
  original_file="$uaa_theirs/jobs/uaa/templates/$file_name"
  if [ -a "$original_file" ]
  then
    vimdiff "$original_file" "$uaa_ours/jobs/uaa/templates/$file_name"
  fi
done
```

LinuxBozo commented:
@rogeruiz so that script doesn't get lost, maybe make it so it only has to ask for theirs path, and commit to this repo?

rogeruiz committed:
Add sync-releases script for easier forkops :fork_and_knife:

LinuxBozo commented on packages/uaa/spec:
```diff
@@ -4,5 +4,5 @@ dependencies:
-- openjdk/openjdk-1.8.0_91-x86_64-mountainlion.tar.gz
-- openjdk/openjdk-1.8.0_91-x86_64-trusty.tar.gz
+- openjdk/openjdk-1.8.0_101-x86_64-mountainlion.tar.gz
+- openjdk/openjdk-1.8.0_101-x86_64-trusty.tar.gz
```
Given: https://gsa-tts.slack.com/archives/cloud-gov-atlas/p1481303821000750, I would expect more changes here.

rogeruiz replied:
right on. I am now seeing a whole bunch of mess. 😞

rogeruiz commented:
Would it be safe to assume that if a deprecated property was removed, that it now needs to be removed from our spec and possibly refactored in our templates?

LinuxBozo commented:
@rogeruiz Bingo. I would expect updates to either cg-deploy manifest or secrets or both.

rogeruiz committed:
Add properties from upstream;

This properties mostly don't exist in our creds/stubs because they're
either new or already refactored. Before this all gets merged in we
still need to evaluate what was added in this commit and make sure they
appear in the credentials and stubs.

rogeruiz committed:
Add in new files from the release

rogeruiz committed:
Update spec and templates from upstream

rogeruiz committed:
Update sync-releases

rogeruiz committed:
Finish up the updates to the spec
